core-java-modules/core-java-collections-maps-7/src/main/java/com/baeldung/map/HashMapWithMaxSizeLimit.java

```diff
@@ -0,0 +1,25 @@
+package com.baeldung.map;
+
+import java.util.HashMap;
+
+public class HashMapWithMaxSizeLimit<K, V> extends HashMap<K, V> {
+
+    private static final long serialVersionUID = 1L;
+
+    private int maxSize = -1;
+    
+   
+    public HashMapWithMaxSizeLimit(int maxSize) {
+        super();
+        this.maxSize = maxSize;
+    }
+
+
+    public V putWithLimit(K key, V value) throws Exception {
```

Comment:
It's better to call this method `put` and then `return super.put(key, value);`. Otherwise, one can still call the "normal" put method of the super call and circumvent the limit.

Reply:
I chose a pushWihLimit because I couldn't make put to throw Exception. Now I realized that I can use unchecked exceptions instead. So, in the new commit, I simply overrided put and as you said, call super.put(key, value).